src/main/java/frc/robot/subsystems/shooter/ShooterIOTalonFX.java

```diff
@@ -0,0 +1,294 @@
+    angleMotorSupplyCurrentStatusSignal = angleMotor.getSupplyCurrent();
+
+    configShootMotors(LEFT_SHOOTER_MOTOR_ID, RIGHT_SHOOTER_MOTOR_ID);
+    configAngleMotor(ANGLE_MOTOR_ID, ANGLE_ENCODER_ID);
+    configDunkerMotor(DUNKER_MOTOR_ID);
+    configDrumMotor(DRUM_MOTOR_ID);
+    this.shooterSensor = new DigitalInput(ShooterConstants.SHOOTER_SENSOR_ID);
+  }
+
+  @Override
+  public void updateInputs(ShooterIOInputs shooterInputs){
+    BaseStatusSignal.refreshAll(
+      shootMotorLeftVelocityStatusSignal,
+      shootMotorRightVelocityStatusSignal,
+      drumMotorVelocityStatusSignal,
+      dunkerMotorVelocityStatusSignal,
+      shootMotorLeftStatorCurrentStatusSignal,
+      shootMotorRightStatorCurrentStatusSignal,
+      drumMotorStatorCurrentStatusSignal,
+      dunkerMotorStatorCurrentStatusSignal,
+      angleMotorStatorCurrentStatusSignal,
+      shootMotorLeftSupplyCurrentStatusSignal,
+      shootMotorRightSupplyCurrentStatusSignal,
+      drumMotorSupplyCurrentStatusSignal,
+      dunkerMotorSupplyCurrentStatusSignal,
+      angleMotorSupplyCurrentStatusSignal
+    );
+    
+    // Updates Left Shooter Motor Inputs
+    shooterInputs.shootMotorLeftStatorCurrentAmps = shootMotorLeftStatorCurrentStatusSignal.getValueAsDouble();
+    shooterInputs.shootMotorLeftSupplyCurrentAmps = shootMotorLeftSupplyCurrentStatusSignal.getValueAsDouble();
+    shooterInputs.shootMotorLeftVelocityRPS = shootMotorLeftVelocityStatusSignal.getValueAsDouble();
+
+    // Updates Right Shooter Motor Inputs
+    shooterInputs.shootMotorRightStatorCurrentAmps = shootMotorRightStatorCurrentStatusSignal.getValueAsDouble();
+    shooterInputs.shootMotorRightSupplyCurrentAmps = shootMotorRightSupplyCurrentStatusSignal.getValueAsDouble();
+    shooterInputs.shootMotorRightVelocityRPS = shootMotorRightVelocityStatusSignal.getValueAsDouble();
+
+    // Updates Angle Motor Inputs
+    shooterInputs.angleMotorStatorCurrentAmps = angleMotorStatorCurrentStatusSignal.getValueAsDouble();
+    shooterInputs.angleMotorSupplyCurrentAmps = angleMotorSupplyCurrentStatusSignal.getValueAsDouble();
+    shooterInputs.angleMotorVelocityRPS = angleMotor.get();
+
+    // Updates Drum Motor Inputs
+    shooterInputs.drumMotorStatorCurrentAmps = drumMotorStatorCurrentStatusSignal.getValueAsDouble();
+    shooterInputs.drumMotorSupplyCurrentAmps = drumMotorSupplyCurrentStatusSignal.getValueAsDouble();
+    shooterInputs.drumMotorVelocityRPS = drumMotorVelocityStatusSignal.getValueAsDouble();
+
+    // Updates Dunker Motor Inputs
+    shooterInputs.dunkerMotorStatorCurrentAmps = dunkerMotorStatorCurrentStatusSignal.getValueAsDouble();
+    shooterInputs.dunkerMotorSupplyCurrentAmps = dunkerMotorSupplyCurrentStatusSignal.getValueAsDouble();
+    shooterInputs.dunkerMotorVelocityRPS = dunkerMotorVelocityStatusSignal.getValueAsDouble();
+
+    // Updates Shooter Sensor
+    shooterInputs.isShooterSensorBlocked = shooterSensor.get();    
+
+  }
```

Comment:
You also need to check all your tunables in this method and update the configuration if any values change. Look at DrivetrainIOCTRE's updateInputs method for an example.